### PR TITLE
Add Bun example

### PR DESCRIPTION
**Pull request by idleberg**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (add or update README or docs)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -41,6 +42,27 @@
...
+
+### Windows
+
+```yaml
+- uses: actions/cache@v4
+  with:
+    path: |
+      ~\.deno
+      ~\AppData\Local\deno
+    key: ${{ runner.os }}-deno-${{ hashFiles('**/deps.ts') }}
```
Comment (joshmgross) [inline examples.md:63]: Does Bun on Windows not have a lockfile?

File: examples.md [hunk 2]
```diff
@@ -41,6 +42,26 @@
 - [Swift - Mint](#swift---mint)
 - [* - Bazel](#---bazel)
 
+## Bun
+
+```yaml
+- uses: actions/cache@v4
+  with:
+    path: |
+      ~/.bun/install/cache
```
Comment (PaulRBerg) [inline examples.md:51]: Why not add `node_modules` to the list of paths, @idleberg @joshmgross?

Comment (RiskyMH) [reply examples.md:51]: @PaulRBerg `node_modules` with Bun is just hardlinks to the aforementioned `~/.bun/install/cache` like PNPM to use less space on disk.

However Bun is usually faster than the cache system, so I wouldn't recommend using it: https://github.com/oven-sh/setup-bun/issues/14#issuecomment-1714116221

Comment (PaulRBerg) [reply examples.md:51]: haha interesting, thanks for your answer!

**Pull request merged**